test/utils/naive_navigator.cpp

```diff
@@ -1,26 +1,51 @@
+#include <numeric>
+
 #include <gtest/gtest.h>
 
 #include <core/types.hpp>
 #include <utils/naive_navigator.hpp>
 
 namespace hyped::test {
 
-void testWithTrajectory(utils::NaiveNavigator &naive_navigator, const core::Trajectory &trajectory)
+void testWithTrajectory(utils::NaiveNavigator &naive_navigator,
+                        const core::RawImuData imu_data,
+                        const core::EncoderData encoder_data,
+                        const core::KeyenceData keyence_data)
 {
-  naive_navigator.update(trajectory);
+  naive_navigator.imuUpdate(imu_data);
+  naive_navigator.encoderUpdate(encoder_data);
+  naive_navigator.keyenceUpdate(keyence_data);
+
+  core::Float sum_imu = 0;
+  for (size_t i = 0; i < core::kNumImus; ++i) {
+    for (size_t j = 0; j < core::kNumAxis; ++j) {
+      sum_imu += imu_data.at(i).at(j);
+    }
+  }
+  core::Float new_acceleration = sum_imu / core::kNumImus;
+  core::Float new_velocity     = 0;
+  core::Float new_displacement
+    = std::accumulate(encoder_data.begin(), encoder_data.end(), 0.0) / core::kNumEncoders;
```

Comment:
Hm, I don't like this test... it's just the code copied and pasted, it doesn't verify any properties.

Reply:
Yeah not quite sure what to do with this since the actual behavior of the class has changed?

Reply:
Think I'm going to make testing this properly something to add to the TODOLater pile.